src/Engine/GameEngine.cs

```diff
@@ -29,6 +29,8 @@ public GameEngine(IOptions<TimeOptions<TTime>> config, ITranslateTime<TTime> tim
 
         public void OnNext<T>(T message, EngineTime time) where T : IEvent
         {
+            if (_disposedValue) throw new ObjectDisposedException("GameEngine");
+
             var timeSubscriptions = _timeSubscriptions[typeof(T)]
```

Comment:
There are probably some considerations regarding multithreading that are missing from the existing code, and therefore probably don't play well into `IDiposable`. I don't have the code open, but I think some of our collections should probably be the `Concurrent*` variants? Just wondering is Disposing (and threads running whilst attempting to dispose) can cause us thread sync/race condition issues